dev/bots/test.dart

```diff
@@ -673,7 +673,7 @@ Future<void> _verifyVersion(String filename) async {
     print('$redLine');
     exit(1);
   }
-  final RegExp pattern = RegExp(r'^[0-9]+\.[0-9]+\.[0-9]+(-pre\.[0-9]+)?$');
+  final RegExp pattern = RegExp(r'[0-9]+\.[0-9]+\.[0-9]+(?:|-pre\.[0-9]+|\+hotfix\.[0-9]+)');
```

Comment:
I'd still switch to the `\d` instead of `[0-9]`.  I've been bit by typos in ranges (e.g. `[0-8]`) before, and it's shorter too:
```suggestion
  final RegExp pattern = RegExp(r'\d+\.\d+\.\d+(?:|-pre\.\d+|\+hotfix\.\d+)');
```

Reply:
fwiw the reason I avoid ranges like \d is that I never know what they actually match. For example, does \d include only ASCII digits or does it include CJK full width digits? Or Arabic digits?

Reply:
EcmaScript standard says "the ten-element set of characters containing the characters 0 through 9 inclusive." So, only ASCII, not Arabic or CKJ.

Reply:
Which is not the same as Perl's `\d`, which is the problem. Looking at the regular expression without context it's impossible to know what it means if it uses `\d`. Looking at it _with_ context you have to be an expert to know what it means (how many people editing this file know if Dart uses PCRE or JSRE?).